build.sbt

```diff
@@ -424,6 +437,14 @@ val mimaFilterSettings = Seq {
     ProblemFilters.exclude[MissingClassProblem]("scala.jdk.package"),
     ProblemFilters.exclude[MissingClassProblem]("scala.jdk.package$"),
 
+    // Artifacts of fixing deprecation warnings
+    ProblemFilters.exclude[IncompatibleMethTypeProblem]("scala.sys.process.ProcessImpl#ThreadProcess.this"),  // private[process]
+    ProblemFilters.exclude[IncompatibleResultTypeProblem]("scala.sys.process.ProcessImpl#PipeSink.sink"),     // private[process]
+    ProblemFilters.exclude[IncompatibleResultTypeProblem]("scala.sys.process.ProcessImpl#PipeSource.source"), // private[process]
+    ProblemFilters.exclude[MissingClassProblem]("scala.collection.immutable.HashMap$KeySet"),                 // private
+    ProblemFilters.exclude[MissingClassProblem]("scala.collection.immutable.HashMap$HashKeySet"),             // private
+    ProblemFilters.exclude[MissingTypesProblem]("scala.concurrent.impl.Promise$Transformation"),              // private[concurrent]
+
```

Comment:
in a possibly vain attempt to avoid a merge conflict with #9232, I put the MiMa excludes for this PR above the MiMa excludes for the previous merged PR